### ISSUE TITLE
Variables-by-ref confusedly trigger SC2034

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#2772]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

demo_by_ref_args()
{
    local -n joined="$1"; shift
    joined="hi there!"
}

declare by_ref
demo_by_ref_args by_ref
echo "$by_ref"
```

#### Here's what shellcheck currently says:
```
Line 6	SC2034: joined appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

Nothing. The output of this script is:

```
hi there!
```

because the `local -n` declaration passes in the argument by reference. When `joined` is assigned, it has effectively been used. Maybe there is another check to create, namely that that `joined` is assigned to something at some point, but 2034 isn't it.

**Comment by pan64** [comment]
duplicate of https://github.com/koalaman/shellcheck/issues/2712

**Comment by brother** [comment]
@pan64 thnx. I agree with that. Closing as dupe.
